consensus/src/common.rs

```diff
@@ -3,15 +3,49 @@
 
 //! Implements common APIs for the blockchain in the context of consensus.
 
+use crate::error::ConsensusError;
 use hex::ToHex;
 use mina_crypto::hash::{Hashable, StateHash};
 use mina_rs_base::consensus_state::ConsensusState;
 use mina_rs_base::global_slot::GlobalSlot;
 use mina_rs_base::protocol_state::{Header, ProtocolState};
+use mina_rs_base::types::{BlockTime, Length};
 
-use crate::checkpoint::is_short_range;
-use crate::density::{relative_min_window_density, ConsensusConstants};
-use crate::error::ConsensusError;
+// TODO: derive from protocol constants
+pub struct ConsensusConstants {
+    /// Point of finality (number of confirmations)
+    pub k: Length,
+    /// Number of slots per epoch
+    pub slots_per_epoch: Length,
+    /// No of slots in a sub-window = 7
+    pub slots_per_sub_window: Length,
+    /// Maximum permissable delay of packets (in slots after the current)
+    pub delta: Length,
+    /// Timestamp of genesis block in unixtime
+    pub genesis_state_timestamp: BlockTime,
+    /// Sub windows within a window
+    pub sub_windows_per_window: Length,
+    /// Number of slots before minimum density is used in chain selection
+    pub grace_period_end: Length,
+}
+
+impl ConsensusConstants {
+    pub fn mainnet() -> Self {
```

Comment:
how about making this a const fn?

Reply:
At this moment, it's not entirely clear how the configs will eventually be loaded. It might also be the case that it may come from a config file.